OrbitCaptureClient/include/OrbitCaptureClient/GpuQueueSubmissionProcessor.h

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_CAPTURE_CLIENT_GPU_QUEUE_SUBMISSION_PROCESSOR_H_
+#define ORBIT_CAPTURE_CLIENT_GPU_QUEUE_SUBMISSION_PROCESSOR_H_
+
+#include <absl/container/flat_hash_map.h>
+
+#include <algorithm>
+#include <functional>
+#include <string>
+#include <vector>
+
+#include "capture.pb.h"
+#include "capture_data.pb.h"
+
+// `GpuQueueSubmission` proto objects come with GPU timestamps (rather than CPU timestamps) for the
```

Comment:
This comment is already a big reason why I prefer the separate class much better now!